lib/env-bundler.js

```diff
@@ -715,30 +721,44 @@ const defaultComponentDescriptorProcessor = function(type, location, code, compo
     });
 }
 
-const assetComponentDescriptorProcessor = function(type, location, code, components) {
+const assetComponentDescriptorProcessor = async function(type, location, code, components) {
   const typePlural = typeToPlural(type);
   const codeExtractor = componentCodeExtractors[type];
 
+  const token = await getToken();
+
   return Promise.all(components.map(async (component) => {
       const code = canonizeCode(codeExtractor(component));
       console.log(`Processing ${type}: ${code}`);
 
-      const split = coreBaseApi.replace('//', '/').split('/');
-      const url = split[0] + '//' + split[1] + component.path;
+      const isProtected = component.group !== 'free';
+      const resourceUrl = isProtected ? apiUrlTable.resourceDetails.replace('false', 'true') : apiUrlTable.resourceDetails;
+      const resourceTypePathSegment = component.type === 'image' ? 'images' : 'documents';
+      const filePathSegment = `${isProtected ? `${component.group}/` : ''}${component.fileName}`;
```

Comment:
this type of ternary operator inside string generation is not recommended since it is harder to understand what is happening here.

Reply:
Agree that it is indeed hard to understand. Will make changes to it.